### DISCUSSION TITLE
How do agents differ (Conversable and assistant)?

### DISCUSSION THREAD

**Discussion by aneeshmb02** [microsoft/autogen#2809] [Q&A]
I was playing around with the sequential chat notebook and I noticed that I get very different results when I use the ConversableAgent and AssistantAgent for the coder agent. Can anyone tell me why is that? Isn't the assistant just a subclass of the conversable?

 I have not configured the assistant in any specific way. The prompt is also pretty generic, just understand a simple code (multiply and square functions) and write unit tests for them. The interesting thing is, the conversable agent actually does a much better job than the assistant.

**Comment by BetoBraga** [comment]
To my mind, Conversable Agent is for textual interactions, while the assistant agent is more focused on execute tasks... But I can be wrong

**Reply by aneeshmb02** [reply]
Well, I am giving the task of writing unit tests, yet the conversable does a better job.

**Reply by BetoBraga** [reply]
Well, in this case, I can't answer you why... Because I'm kinda new to autogen too - but, I would guess it's because writing unit tests still textual task... And I would guess, again, that assistantagent is more for executing these texts

**Reply by aneeshmb02** [reply]
If you mean execution of code, isn't that done by the userproxy agent? Assistant is just supoosed to write code (for unit testing in my case).

**Reply by ChinmayBH** [reply]
Hello, 
ConversableAgent is a class for generic conversable agents which can be configured as assistant or user proxy.
so there might be a good results with conversable as all three inherited from same class.
